### ISSUE TITLE
Conflict with vim-plug in Windows

### ISSUE THREAD

**Issue by xvim64** [vim/vim#9580]
### Steps to reproduce

run gvim.exe

### Expected behaviour

Load vim-plug without conflict

### Version of Vim

8.2.4167

### Environment

Windows 10 64-bit 21H2

### Logs and stack traces

```shell
[ error message at gvim startup]
Error detected while processing C:\home\.vimrc[2806]..C:\home\.vim\vim-plug\plug.vim:
line  709:
E129: Function name required
line  710:
E121: Undefined variable: a:0
line  711:
E133: :return not inside a function
Error detected while processing C:\home\.vimrc[2806]..C:\home\.vim\vim-plug\plug.vim[726]..C:\home\.vimrc[2806]..C:\home\.vim\vim-plug\plug.vim:
line  725:
E133: :return not inside a function

[ .vimrc ]
2806: source ~/.vim/vim-plug/plug.vim
2807: call plug#begin('~/.vim/plugged')
2808:    Plug 'vim-airline/vim-airline'
2809:    Plug 'vim-airline/vim-airline-themes'
2810:    Plug 'neoclide/coc.nvim', {'branch': 'release'}
2811: call plug#end()
```

**Comment by lacygoill** [comment]
This is a regression introduced in patch [8.2.4163](https://github.com/vim/vim/releases/tag/v8.2.4163).  It will be fixed.

In the meantime, either downgrade your Vim to an older version, or apply this patch to your `autoload/plug.vim` script:
```diff
diff --git a/plug.vim b/plug.vim
index 8a195d3..91ba1a7 100644
--- a/plug.vim
+++ b/plug.vim
@@ -263,7 +263,7 @@ function! plug#begin(...)
 endfunction
 
 function! s:define_commands()
-  command! -nargs=+ -bar Plug call plug#(<args>)
+  command! -nargs=+ -bar Plug call plug#_(<args>)
   if !executable('git')
     return s:err('`git` executable not found. Most commands will not be available. To suppress this message, prepend `silent!` to `call plug#begin(...)`.')
   endif
@@ -706,7 +706,7 @@ function! s:lod_map(map, names, with_prefix, prefix)
   call feedkeys(substitute(a:map, '^<Plug>', "\<Plug>", '') . extra)
 endfunction
 
-function! plug#(repo, ...)
+function! plug#_(repo, ...)
   if a:0 > 1
     return s:err('Invalid number of arguments (1..2)')
   endif
```

**Comment by brammool** [comment]
fixed with patch 8.2.4167.  It also adds a test for using an empty name in legacy script.

**Comment by k-takata** [comment]
> fixed with patch 8.2.4167.

Maybe a mistake of 8.2.4168?

**Comment by lacygoill** [comment]
Yes, indeed, the fix to restore autoload functions with no name was implemented in patch [8.2.4168](https://github.com/vim/vim/releases/tag/v8.2.4168).

**Comment by brammool** [comment]
Only off by 0.024 %  :-)
